typescript/deploy/src/deploy.ts

```diff
@@ -48,17 +50,33 @@ export abstract class AbacusDeployer<
 
   abstract deployContracts(chain: Chain, config: Config): Promise<Contracts>;
 
-  async deploy() {
-    this.logger('Start Deploy');
-    this.verificationInputs = objMap(this.configMap, () => []);
-    const chains = this.multiProvider.chains();
-    const entries: [Chain, Contracts][] = [];
-    for (const chain of chains) {
+  async deploy(
+    partialDeployment: Partial<Record<Chain, Contracts>> = this
+      .deployedContracts,
+  ): Promise<Record<Chain, Contracts>> {
+    const deployedChains = Object.keys(partialDeployment);
+    const configChains = Object.keys(this.configMap);
+    const targetChains = this.multiProvider
+      .chains()
+      .filter(
+        (chain) =>
+          configChains.includes(chain) && !deployedChains.includes(chain),
+      );
+    this.logger(
+      `Start deploy to ${targetChains} ${
+        deployedChains.length > 0
+          ? `(already deployed to ${deployedChains})`
+          : ''
+      }`,
+    );
+    for (const chain of targetChains) {
       this.logger(`Deploying to ${chain}...`);
-      const result = await this.deployContracts(chain, this.configMap[chain]);
-      entries.push([chain, result]);
+      this.deployedContracts[chain] = await this.deployContracts(
+        chain,
+        this.configMap[chain],
+      );
     }
-    return Object.fromEntries(entries) as Record<Chain, Contracts>;
+    return this.deployedContracts as Record<Chain, Contracts>;
```

Comment:
I don't know what I'm missing here, but how does `deployedContracts` receive the data from `partialDeployment`?